### DISCUSSION TITLE
parallel bootstrapping

### DISCUSSION THREAD

**Discussion by kmytsay** [JeschkeLab/DeerLab#376] [General] [answered]
The bootstrapping analysis seems to be fitting one sample at a time, instead of fitting all the samples together.  This process is very slow, is there a way to make it faster?

**Comment by luisfabib** [comment, marked as answer]
The `bootstrap_analysis` function has a `cores` option implemented to do multiprocessing of the bootstrap evaluation. Sadly I just realized that the `fit` function does not take that as an option. 

We will have to implement it for the next patch.

**Reply by kmytsay** [reply]
Awesome! Can't wait for bootstrapping to be faster.

**Reply by kmytsay** [reply]
Is there a way to implement bootstrapping not through the fit function in the current set up?

**Reply by luisfabib** [reply]
You can edit the `model.py` file of the DeerLab folder (which can be found in the Python installation of your system) and edit line `1138` from:
```
param_uq = bootstrap_analysis(bootstrap_fcn,ysplit,fitresults.model,samples=bootstrap,noiselvl=noiselvl)
```
to 
```
param_uq = bootstrap_analysis(bootstrap_fcn,ysplit,fitresults.model,samples=bootstrap,noiselvl=noiselvl, cores=-1)
```
To use the maximum number of cores in your computer during bootstrapping. This should give you the functionality until we release the patch with that same edit.

**Reply by kmytsay** [reply]
Great! thank you!

**Reply by kmytsay** [reply]
I have a similar question about the compactness penalty.  It seems that the Non-linear least-squares routine runs until an optimized compactness parameter is reached.  This process is quite slow, is it possible for the fitting routine to fit all compactness parameter at the same time?
